syntax = "proto3";

option java_package = "ex.grpc";

package mutex;

// Defines the service
// need to check: can I do multiple rpc's or services?
service mutex_service {
    // Function invoked to send the request
    rpc send_mutex_request (mutex_request) returns (mutex_reply) {};
	rpc send_deferred_reply (mutex_deferred_reply) returns (mutex_reply) {};
}


message mutex_request {
    int32 lockno = 1;
    int32 sequence_no = 2;
	int32 id = 3;
}

message mutex_reply {
    int32 acknowledgement = 1;
}

message mutex_deferred_reply {
	int32 sender = 1;
	int32 lockno = 2;
}
